front/lib/api/assistant/configuration.ts

```diff
@@ -37,48 +37,79 @@ export async function getAgentConfiguration(
     throw new Error("Cannot find AgentConfiguration: no workspace.");
   }
   const agent = await AgentConfiguration.findOne({
+    //logging: console.log,
     where: {
       sId: agentId,
       workspaceId: owner.id,
     },
+    include: [
+      {
+        model: AgentGenerationConfiguration,
+        as: "generationConfiguration",
+      },
+      {
+        model: AgentRetrievalConfiguration,
+        as: "retrievalConfiguration",
+        include: [
+          {
+            model: AgentDataSourceConfiguration,
```

Comment:
I think I would do these manually. Since it's a sub-array it has to be another query so we can do it ourselves?

get `generation` and `action` through join but get the AgentDataSourceConfiguration manually but joining in the dataSource and workspace

Reply:
Done!